pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -102,103 +102,98 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 
-		// get the clusterRoleName by spliting the clusterRoleBindng name
-		clusterRBName := strings.Split(clusterRoleBindingName, "-")
-		clusterRoleName := clusterRBName[0]
-		subjectName := clusterRBName[1]
-
-		// create a new clusterRoleBinding on cluster
-		newCRB := newClusterRoleBinding(clusterRoleName, subjectName, instance.Spec.SubjectKind)
-		err := r.client.Create(context.TODO(), newCRB)
-		if err != nil {
 			var clusterRoleNames []string
-			clusterRoleNames = append(clusterRoleNames, clusterRoleName)
+			clusterRoleNames = append(clusterRoleNames, crClusterRoleName)
 
-			// update the condition if creation of a ClusterRoleBinding has failed
-			instance := controllerutil.UpdateCondition(instance, "Unable to create ClusterRoleBinding: "+err.Error(), clusterRoleNames, true, managedv1alpha1.SubjectPermissionFailed)
+			// helper func to update the condition of the SubjectPermission object
+			instance := controllerutil.UpdateCondition(instance, crClusterRoleName+" for clusterPermission does not exist", clusterRoleNames, true, "Failed")
 			err = r.client.Status().Update(context.TODO(), instance)
 			if err != nil {
-				reqLogger.Error(err, "Failed to update condition.")
+				reqLogger.Error(err, "Failed to update condition. 3")
 				return reconcile.Result{}, err
 			}
-			reqLogger.Error(err, "Failed to create clusterRoleBinding")
-			return reconcile.Result{}, err
+			return reconcile.Result{}, nil
 		}
 
-		// instead of updating the condition just log each successfully created ClusterRoleBinding
-		reqLogger.Info(fmt.Sprintf("Successfully created ClusterRoleBinding %s", clusterRoleName))
+		// build a clusterRoleBindingNameList which consists of clusterRoleName-subjectName
+		crClusterRoleBindingNameList := buildClusterRoleBindingCRList(instance)
 
-		// Add Prometheus metrics for this CR
-		localmetrics.AddPrometheusMetric(instance)
-		return reconcile.Result{}, nil
-	}
+		// check ClusterRoleBindingName
+		populateCrClusterRoleBindingNameList := populateClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
 
-	// update condition if all ClusterRoleBindings added succesfully
-	instance = controllerutil.UpdateCondition(instance, "Successfully created all ClusterRoleBindings", populateCrClusterRoleBindingNameList, true, managedv1alpha1.SubjectPermissionCreated)
+		// loop through crClusterRoleBindingNameList
+		for _, clusterRoleBindingName := range populateCrClusterRoleBindingNameList {
 
-	// get the NamespaceList
-	nsList := &corev1.NamespaceList{}
-	opts = client.ListOptions{Namespace: request.Namespace}
-	err = r.client.List(context.TODO(), &opts, nsList)
-	if err != nil {
-		reqLogger.Error(err, "Failed to get clusterRoleBindingList")
-		return reconcile.Result{}, err
-	}
+			// get the clusterRoleName by spliting the clusterRoleBindng name
+			clusterRBName := strings.Split(clusterRoleBindingName, "-")
+			clusterRoleName := clusterRBName[0]
+			subjectName := clusterRBName[1]
```

Comment:
Seems like there might be a better way to do this? cc/ @fahlmant

Reply:
Yes. The most recent plan was to add these as annotations to the clusterrolebindings on creation

Reply:
Yes I agree. This will be implemented in the next PR, separated from this one as it is pretty big.

Reply:
No sure about annotations, lets merge as is and we'll discuss it.